### PR TITLE
test: Cypress - added tags - @tag.JS, @tag.Perf, @tag.Settings,  @tag.PropertyPane,  @tag.Theme

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
- cypress/e2e/Regression/ClientSide/PropertyPane
- cypress/e2e/Regression/ClientSide/ThemingTests
- cypress/e2e/Regression/ClientSide/Refactoring
- cypress/e2e/Regression/ClientSide/ProductRamps
- cypress/e2e/Regression/ClientSide/Performance
- cypress/e2e/Regression/ClientSide/PeekOverlay

#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [ ] Manual
- [ ] Cypress

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **Tests**
	- Enhanced test categorization with the addition of tags for better filtering and identification across various test suites.
	- Improved readability and maintenance of test cases through reformatting and restructuring.
	- Updated test suite options for improved test coverage and identification.

- **Refactor**
	- Optimized test setup and execution logic for several feature-related test suites.

- **Bug Fixes**
	- Adjusted test cases to accurately reflect application behavior, particularly in toast message validation and property setting scenarios.

- **Documentation**
	- Added comments for clarity in test code to aid in understanding test case intent and functionality.

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Performance/LinkRelPreload_Spec.js [hunk 1]
```diff
@@ -27,28 +27,32 @@ import emptyDSL from "../../../../fixtures/emptyDSL.json";
...
-    cy.get("#sidebar").should("be.visible");
-
-    _.deployMode.DeployApp();
-
-    testPreloadMetadata("view-mode");
-  });
-});
+describe(
+  "html should include preload metadata for all code-split javascript",
+  { tags: ["@tag.Perf"] },
```
Comment (mohanarpit) [inline app/client/cypress/e2e/Regression/ClientSide/Performance/LinkRelPreload_Spec.js:32]: Can we avoid the tag "performance"? Performance is a meta-quality of the product and not a feature in the product. Let's keep tags aligned towards product features. This way, as an engineer, I can decide "I changed X part of the product, hence I should test with X tag".

Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/Performance/LinkRelPreload_Spec.js:32]: Shall I tag this a sanity. This is the comment in the test case
```
This test exists to ensure that all code-split chunks are <link rel="preload">ed. If this test
started failing for you, it’s likely you import()ed some new chunks that the edit or the view mode uses.
To fix the test, see preloading instructions in public/index.html.
```
I am not sure how to tag this correctly. Suggestion please?

Comment (mohanarpit) [reply app/client/cypress/e2e/Regression/ClientSide/Performance/LinkRelPreload_Spec.js:32]: Can you tag this part of the IDE? It seems to be relevant to index.html.

Comment (sharanya-appsmith) [reply app/client/cypress/e2e/Regression/ClientSide/Performance/LinkRelPreload_Spec.js]: Done.

**Review by Aishwarya-U-R — APPROVED** [review]
Proxy Approval.

**Pull request merged**